internal/util/git/git.go

```diff
@@ -174,3 +193,70 @@ func NormalizeGitUrl(gitUrl string) string {
 	normalizedUrl = strings.Trim(normalizedUrl, "/")
 	return normalizedUrl
 }
+
+// UpdateOptionsWithGitConfig updates the given clone or fetch options object with
+// information loaded from the git config found based on the given config scope.
+// The function only takes into account HTTP URLs (not SSH).
+func UpdateOptionsWithGitConfig[T git.CloneOptions | git.FetchOptions | git.PullOptions](
+	scope config.Scope, options *T, repoURL string,
+) error {
+	gitConfig, err := config.LoadConfig(scope)
+	if err != nil {
+		return fmt.Errorf("error loading git config: %v", err)
+	}
+
+	// Check if LoadConfig resulted in an empty/new config (maybe because the git config file was not setup)
+	if len(gitConfig.URLs) == 0 && gitConfig.Raw == nil {
+		return nil
+	}
+
+	httpSubSecKey := ""
+
+	rURL, err := url.Parse(repoURL)
+	if err != nil {
+		return fmt.Errorf("error parsing repo URL '%s': %v", repoURL, err)
+	}
+
+	switch any(*options).(type) {
+	case git.CloneOptions:
+		insteadOf := fmt.Sprintf("%s://%s", rURL.Scheme, rURL.Host)
+		for k, v := range gitConfig.URLs {
+			if v.InsteadOf == insteadOf {
+				keyURL, err := url.Parse(k)
+				if err != nil {
+					return fmt.Errorf("invalid URL '%s' in git config: %v", k, err)
+				}
+
+				any(options).(*git.CloneOptions).URL = fmt.Sprintf("%s%s", k, rURL.Path)
+				httpSubSecKey = fmt.Sprintf("%s://%s", keyURL.Scheme, keyURL.Host)
+				break
+			}
+		}
+	case git.FetchOptions, git.PullOptions:
+		httpSubSecKey = fmt.Sprintf("%s://%s", rURL.Scheme, rURL.Host)
```

Comment:
I see `git.FetchOptions` and `git.PullOptions` both have a `remoteURL` field in them - want to verify we don't need to set that, do we?

Reply:
With my initial tests, I was passing the remote URL for the fetch options update. After making adjustments due to merge conflicts, I think it would be beneficial to set the RemoteURL field. I'm working on testing that now.